vm-deployment/vm.bicep

```
@@ -0,0 +1,282 @@
+  }
+}
+
+// Virtual machine.
+resource virtualMachines_resource 'Microsoft.Compute/virtualMachines@2024-03-01' = {
+  name: virtualMachines
+  location: location
+  zones: [
+    '1'
+  ]
+  properties: {
+    hardwareProfile: {
+      vmSize: vmSize
+    }
+    additionalCapabilities: {
+      hibernationEnabled: false
+    }
+    storageProfile: {
+      imageReference: {
+        publisher: 'canonical'
+        offer: '0001-com-ubuntu-server-focal'
+        sku: '20_04-lts-gen2'
+        version: 'latest'
+      }
+      osDisk: {
+        osType: 'Linux'
+        name: '${virtualMachines}_OsDisk_1_bcb7e14b83d24fc7a077197bdf465401'
+        createOption: 'FromImage'
+        caching: 'ReadWrite'
+        managedDisk: {
+          storageAccountType: 'Premium_LRS'
+        }
+        deleteOption: 'Delete'
+        diskSizeGB: 30
+      }
+      dataDisks: []
+      diskControllerType: 'SCSI'
+    }
+    osProfile: {
+      computerName: virtualMachines
+      adminUsername: virtualMachines
+      linuxConfiguration: {
+        disablePasswordAuthentication: true
+        ssh: {
+          publicKeys: [
+            {
+              path: '/home/${virtualMachines}/.ssh/authorized_keys'
+              keyData: sshPublicKey 
+            }
+          ]
+        }
+        provisionVMAgent: true
+        patchSettings: {
+          patchMode: 'AutomaticByPlatform'
+          automaticByPlatformSettings: {
+            rebootSetting: 'IfRequired'
+            bypassPlatformSafetyChecksOnUserSchedule: false
+          }
+          assessmentMode: 'ImageDefault'
+        }
+      }
+      secrets: []
+      allowExtensionOperations: true
+    }
+    securityProfile: {
+      uefiSettings: {
+        secureBootEnabled: true
+        vTpmEnabled: true
+      }
+      securityType: 'TrustedLaunch'
+    }
+    networkProfile: {
+      networkInterfaces: [
+        {
+          id: networkInterfaces_resource.id
+          properties: {
+            deleteOption: 'Detach'
+          }
+        }
+      ]
+    }
+    diagnosticsProfile: {
+      bootDiagnostics: {
+        enabled: true
+      }
+    }
+  }
+}
+
+// Subnet.
+resource virtualNetworks_default 'Microsoft.Network/virtualNetworks/subnets@2023-11-01' = {
+  name: 'default'
+  parent: virtualNetworks_resource
+  properties: {
+    addressPrefix: subnetAddressPrefix
+    delegations: []
+    privateEndpointNetworkPolicies: 'Disabled'
+    privateLinkServiceNetworkPolicies: 'Enabled'
+  }
+}
+
+// Network interface.
+resource networkInterfaces_resource 'Microsoft.Network/networkInterfaces@2023-11-01' = {
+  name: networkInterfaces
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig1'
+        properties: {
+          privateIPAddress: privateIPAddress
+          privateIPAllocationMethod: 'Dynamic'
+          publicIPAddress: {
+            id: publicIPAddresses_resource.id
+            properties: {
+              deleteOption: 'Detach'
+            }
```

```suggestion
            id: publicIPAddress.id
```
I'm not sure - what happens if we reference a resource (by `id`) and change it's `properties` at the same time? Isn't `deleteOption=Detach` set by default?